roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,104 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+
+class DockerImageAvailability(OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        is_containerized = get_var(task_vars, "openshift", "common", "is_containerized")
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+        deployment_type = get_var(task_vars, "deployment_type")
+
+        is_origin_deployment = deployment_type == "origin"
+
+        changed = False
+        failed_pulls = set()
+
+        required_images = self.rpm_docker_images(is_origin_deployment)
+        if is_containerized:
+            required_images = self.containerized_docker_images(is_origin_deployment)
+
+        pulled_images = self.pull_images(self.add_image_tags(required_images, openshift_release), tmp, task_vars)
+
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
```

Comment:
Ah, I wrote this check under the assumption that `openshift_image_tag` could be optionally set in the config (so there would almost certainly be cases where this is not set). If it was the case that it was not set, then I would not be able to pull images with qualified tags, so I failed before getting to that part.

However it looks like I can count on `openshift_image_tag` to be set by the time this module runs, so I will remove that block.